### PR TITLE
feat: (e2e) onboarding tests and quick login method

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
- Adds End to End tests for the Onboarding Flow
    - Full Flow
    - User Skips the onboarding Flow
- Adds a method that signs up and logs in a user for writing further tests easily!

## Type of change

- [x] Enhancement (small improvements)

## How should this be tested?
Test the test hmm i c ;)

1. `docker compose --env-file /dev/null up`
2. `pnpm test:e2e`

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://formbricks.com/clmyhzfrymr4ko00hycsg1tvx) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/playwright/utils.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  let roles = ["Project Manager", "Engineer", "Founder", "Marketing Specialist"];
+  let useCases = [
+    "Increase conversion",
+    "Improve user retention",
+    "Increase user adoption",
+    "Sharpen marketing messaging",
+    "Support sales",
+  ];
+  const productName = randomBytes(8).toString("hex");
+  const role = roles[Math.floor(Math.random() * roles.length)];
```
Comment (mattinannt) [inline apps/web/playwright/utils.ts:21]: I'm not sure if we should use randomness here as it makes the test non-deterministic.
This is what I got from chatGPT for Anshumans PR where he also used randomness for unit tests:

---

Using random values in unit tests, like your `randBrowser()` and `randBoolean()` functions, can lead to non-deterministic tests, which might not be ideal in many testing scenarios. Here's a breakdown of the considerations:

### Non-Deterministic Tests: Drawbacks

1. **Reproducibility**: If a test fails due to a specific random value, it can be challenging to reproduce and debug the issue since the same value may not be generated again.
2. **Flakiness**: Tests might pass sometimes and fail at others, depending on the random values generated. This can lead to uncertainty about the stability and reliability of the code.
3. **Coverage Inconsistency**: Random values might not consistently cover all the edge cases or paths in the code, leading to inconsistent test coverage.

### Deterministic Tests: Advantages

1. **Reproducibility**: Tests produce the same results every time, making it easier to identify and fix issues.
2. **Reliability**: Consistent outcomes increase confidence in the test results and the code's stability.
3. **Complete Coverage**: You can design tests to cover all relevant cases and edge cases, ensuring comprehensive testing.

### When to Use Randomness

Using randomness in tests can be beneficial in certain scenarios:

1. **Stress Testing**: To test how the system behaves under varied and unpredictable inputs.
2. **Fuzz Testing**: To identify potential crashes or unexpected behavior in the system.
3. **Large Data Sets**: When dealing with huge datasets, random sampling might be necessary for practicality.

### Best Practices

- **Controlled Randomness**: If you use random values, consider using a seeded random number generator. This way, the "random" values are the same across different test runs, maintaining some level of determinism.
- **Specific Test Cases for Edge Conditions**: Write separate deterministic tests for edge cases and specific conditions you want to ensure are always tested.
- **Monitoring Flaky Tests**: If you use randomness, monitor for flakiness in your test suite, and investigate tests that show inconsistent results.

### Conclusion

In your context, if the random values are not creating flakiness or missing coverage, they might be acceptable. However, generally, it's better to have deterministic tests, especially for unit testing, where you're testing specific functions or components under known conditions. For broader system testing or in specific scenarios like stress or fuzz testing, randomness can be more appropriate.

**Review by mattinannt — COMMENTED** [review]
@ShubhamPalriwala looks great! 💪
I only have one concern regarding randomness

File: apps/web/playwright/utils.ts (hunk 1 above)
Comment (ShubhamPalriwala) [reply apps/web/playwright/utils.ts]: Hey, thanks for this! I have removed the randomness now, please take a look 🤞🏼

**Review by mattinannt — APPROVED** [review]
@ShubhamPalriwala thank you, ready to merge :-) 💪🚀
